### DISCUSSION TITLE
How to change Input Request MediaType

### DISCUSSION THREAD

**Discussion by KiraPC** [fastapi/fastapi#7007] [Questions] [answered]
Hi Guys,

Is there a way to change the input media type accepted for a specific route?

E.G.

I would like to set the route /foo/bar to accept only the media type 'application/octet-stream', and i would like also that this is reflected to the openapi template.

Something like the "consume" tag in Spring.

**Comment by valentin994** [comment]
You could fetch that in the request header and then check if it is the right one.

**Comment by valentin994** [comment, marked as answer]
> Hi Guys,
> 
> Is there a way to change the input media type accepted for a specific route?
> 
> E.G.
> 
> I would like to set the route /foo/bar to accept only the media type 'application/octet-stream', and i would like also that this is reflected to the openapi template.
> 
> Something like the "consume" tag in Spring.

One approach is this one below, or you could add middleware that processes the headers first.

```
@app.get("/req")
async def get_header(request: Request):
    print(request.headers)
    return {"do with them what you want"}
```

**Comment by KiraPC** [comment]
Yes, thanks, I just arrived after opened this issue.

**Comment by tiangolo** [comment]
Thanks for the help here @valentin994 ! 👏  🙇 

Thanks for reporting back and closing the issue @KiraPC  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
